app/services/licence.service.js

```diff
@@ -0,0 +1,59 @@
+'use strict'
+
+/**
+ * @module InvoiceService
+ */
+
+const { LicenceModel } = require('../models')
+
+class LicenceService {
+  /**
+  * Accepts a transaction and creates an entry in the Licences table if one doesn't already exist for the transaction's
+  * invoice/licence/customer ref/financial year. It updates the count and value stats as per the transaction details
+  * and returns the resulting licence.
+  *
+  * Note that the updated stats are _not_ saved back to the database; it is up to the caller to do this.
+  *
+  * @param {Object} A transaction object
+  * @returns {Object} A licence object
+  */
+
+  static async go (transaction) {
```

Comment:
The only reason for possible changes is if we take any of my suggestions in the migration script on board. If so this will change the underlying table so some tweaks would be needed.